File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> ans;
    void inorderTraversalUtil(TreeNode* root) {
        if (root) {
            this->inorderTraversalUtil(root->left);
            ans.push_back(root->val);
            this->inorderTraversalUtil(root->right);
        }
    }
    vector<int> inorderTraversal(TreeNode* root) {
        this->inorderTraversalUtil(root);
        return ans;
    }
};

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> ans = new ArrayList<>();
    public void inorderTraversalUtil(TreeNode root) {
        if (root != null) {
            this.inorderTraversalUtil(root.left);
            ans.add(root.val);
            this.inorderTraversalUtil(root.right);
        }
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        this.inorderTraversalUtil(root);
        return ans;
    }
}